File: dailyHOTEL/app/src/main/java/com/daily/dailyhotel/screen/home/stay/inbound/detail/StayDetailAnalyticsImpl.java
package com.daily.dailyhotel.screen.home.stay.inbound.detail;

import android.app.Activity;

import com.daily.base.util.DailyTextUtils;
import com.daily.dailyhotel.entity.StayBookDateTime;
import com.daily.dailyhotel.entity.StayDetail;
import com.daily.dailyhotel.parcel.analytics.StayDetailAnalyticsParam;
import com.daily.dailyhotel.parcel.analytics.StayPaymentAnalyticsParam;
import com.twoheart.dailyhotel.util.analytics.AnalyticsManager;

import java.util.HashMap;
import java.util.Map;

public class StayDetailAnalyticsImpl implements StayDetailPresenter.StayDetailAnalyticsInterface
{
    private StayDetailAnalyticsParam mAnalyticsParam;

    @Override
    public void setAnalyticsParam(StayDetailAnalyticsParam analyticsParam)
    {
        mAnalyticsParam = analyticsParam;
    }

    @Override
    public StayDetailAnalyticsParam getAnalyticsParam()
    {
        return mAnalyticsParam;
    }

    @Override
    public void onScreen(Activity activity, StayBookDateTime stayBookDateTime, StayDetail stayDetail, int priceFromList)
    {

    }

    @Override
    public void onScreenRoomList(Activity activity)
    {
        if (activity == null)
        {
            return;
        }
    }

    @Override
    public void onEventShareKakaoClick(Activity activity, boolean login, String userType, boolean benefitAlarm, int gourmetIndex, String gourmetName)
    {

    }

    @Override
    public void onEventShareSmsClick(Activity activity, boolean login, String userType, boolean benefitAlarm, int gourmetIndex, String gourmetName)
    {

    }

    @Override
    public void onEventDownloadCoupon(Activity activity, String stayName)
    {

    }

    @Override
    public void onEventDownloadCouponByLogin(Activity activity, boolean login)
    {

    }

    @Override
    public void onEventShare(Activity activity)
    {

    }

    @Override
    public void onEventHasHiddenMenus(Activity activity)
    {

    }

    @Override
    public void onEventChangedPrice(Activity activity, boolean deepLink, String stayName, boolean soldOut)
    {

    }

    @Override
    public void onEventCalendarClick(Activity activity)
    {

    }

    @Override
    public void onEventOrderClick(Activity activity, StayBookDateTime stayBookDateTime, String stayName, String menuName, String category, int discountPrice)
    {

    }

    @Override
    public void onEventScrollTopMenuClick(Activity activity, String stayName)
    {

    }

    @Override
    public void onEventMenuClick(Activity activity, int menuIndex, int position)
    {

    }

    @Override
    public void onEventTrueReviewClick(Activity activity)
    {

    }

    @Override
    public void onEventMoreMenuClick(Activity activity, boolean opened, int stayIndex)
    {

    }

    @Override
    public void onEventImageClick(Activity activity, String stayName)
    {

    }

    @Override
    public void onEventConciergeClick(Activity activity)
    {

    }

    @Override
    public void onEventMapClick(Activity activity, String stayName)
    {

    }

    @Override
    public void onEventClipAddressClick(Activity activity, String stayName)
    {

    }

    @Override
    public void onEventWishClick(Activity activity, StayBookDateTime stayBookDateTime, StayDetail stayDetail, int priceFromList, boolean myWish)
    {

    }

    @Override
    public void onEventCallClick(Activity activity)
    {

    }

    @Override
    public void onEventFaqClick(Activity activity)
    {

    }

    @Override
    public void onEventHappyTalkClick(Activity activity)
    {

    }
}

File: dailyHOTEL/app/src/main/java/com/daily/dailyhotel/screen/home/stay/inbound/detailk/StayDetailAnalyticsImpl.kt
package com.daily.dailyhotel.screen.home.stay.inbound.detailk

import android.app.Activity
import com.daily.dailyhotel.entity.StayBookDateTime
import com.daily.dailyhotel.entity.StayDetailk
import com.daily.dailyhotel.entity.StayRoom
import com.daily.dailyhotel.parcel.analytics.StayDetailAnalyticsParam
import com.daily.dailyhotel.parcel.analytics.StayPaymentAnalyticsParam

class StayDetailAnalyticsImpl : StayDetailInterface.AnalyticsInterface {

    private var analyticsParam: StayDetailAnalyticsParam? = null

    override fun setAnalyticsParam(analyticsParam: StayDetailAnalyticsParam) {
        this.analyticsParam = analyticsParam
    }

    override fun getStayPaymentAnalyticsParam(stayDetail: StayDetailk, stayRoom: StayRoom): StayPaymentAnalyticsParam {

        return StayPaymentAnalyticsParam()
    }

    override fun onScreen(activity: Activity, stayBookDateTime: StayBookDateTime, stayDetail: StayDetailk?, priceFromList: Int) {
    }

    override fun onScreenRoomList(activity: Activity, stayBookDateTime: StayBookDateTime, stayDetail: StayDetailk, priceFromList: Int) {
    }

    override fun onEventRoomListOpenClick(activity: Activity, stayName: String) {
    }

    override fun onEventRoomListCloseClick(activity: Activity, stayName: String) {
    }

    override fun onEventRoomClick(activity: Activity, roomName: String) {
    }

    override fun onEventShareKakaoClick(activity: Activity, login: Boolean, userType: String, benefitAlarm: Boolean, stayIndex: Int, stayName: String?) {
    }

    override fun onEventLinkCopyClick(activity: Activity) {
    }

    override fun onEventMoreShareClick(activity: Activity) {
    }

    override fun onEventDownloadCoupon(activity: Activity, stayName: String?) {
    }

    override fun onEventDownloadCouponByLogin(activity: Activity, login: Boolean) {
    }

    override fun onEventShare(activity: Activity) {
    }

    override fun onEventChangedPrice(activity: Activity, deepLink: Boolean, stayName: String, soldOut: Boolean) {
    }

    override fun onEventCalendarClick(activity: Activity) {
    }

    override fun onEventBookingClick(activity: Activity, stayBookDateTime: StayBookDateTime, stayIndex: Int,
                                     stayName: String, roomName: String, discountPrice: Int, category: String,
                                     provideRewardSticker: Boolean, isOverseas: Boolean) {
    }

    override fun onEventTrueReviewClick(activity: Activity) {
    }

    override fun onEventTrueVRClick(activity: Activity, stayIndex: Int) {
    }

    override fun onEventImageClick(activity: Activity, stayName: String?) {
    }

    override fun onEventConciergeClick(activity: Activity) {
    }

    override fun onEventMapClick(activity: Activity, stayName: String?) {
    }

    override fun onEventClipAddressClick(activity: Activity, stayName: String?) {
    }

    override fun onEventWishClick(activity: Activity, stayBookDateTime: StayBookDateTime, stayDetail: StayDetailk, priceFromList: Int, myWish: Boolean) {
    }

    override fun onEventCallClick(activity: Activity) {
    }

    override fun onEventFaqClick(activity: Activity) {
    }

    override fun onEventHappyTalkClick(activity: Activity) {
    }

    override fun onEventShowTrueReview(activity: Activity, stayIndex: Int) {
    }

    override fun onEventShowCoupon(activity: Activity, stayIndex: Int) {
    }

    override fun onEventTrueAwards(activity: Activity, stayIndex: Int) {
    }

    override fun onEventTrueAwardsClick(activity: Activity, stayIndex: Int) {
    }

}